tools/config/__init__.py

```diff
@@ -696,7 +748,8 @@ def _assign_new_offset(rom_start, start, new_offset, region_name):
                 "Can not place % region inside previous region" % region_name)
         return newstart
 
-    def _generate_bootloader_build(self, rom_start, rom_size):
+    def _generate_bootloader_build(self, active_rom_regions):
```

Comment:
This will have to become:
```suggestion
    def _generate_bootloader_build(self, ROM=None, **other_rom_memories):
```

Reply:
It will become, 
 `def _generate_bootloader_build(self, **other_rom_memories):` , right? Why `ROM=None`. I am only passing the dictionary.

Reply:
So that you can use `ROM` in the next diff.

Reply:
May be I am misunderstanding, or missing something. We do not have ROM property anymore, and given the fact that each target will have IROM1 or program flash defined, we will be passing the dictionary with ROM defined for sure. So for the next diff, instead of
`rom_start, rom_size = next(iter(active_rom_regions.values()))`
we can do
`rom_start, rom_size = active_rom_regions.get('ROM')`
It's very well possible that I am missing some high level view of the intention behind this function. Is it? Let me commit my latest changes and lets get back to this if needed.

Reply:
> We do not have ROM property anymore

That's unrelated.

> given the fact that each target will have IROM1 or program flash defined, we will be passing the dictionary with ROM defined for sure

This make the dictionary item `ROM` a local.

Nothing "high level" or anything like that. Feel free to implement it the way you're suggesting, with the modification `rom_start, rom_size = active_rom_regions['ROM']`.